Core/DISET/private/Transports/SSL/SocketInfoFactory.py

```diff
@@ -54,8 +55,12 @@ def __socketConnect( self, hostAddress, timeout, retries = 2 ):
     return S_ERROR( ", ".join( errs ) )
 
   def __sockConnect( self, hostAddress, sockType, timeout, retries ):
-    osSocket = socket.socket( sockType, socket.SOCK_STREAM )
-    #osSocket.setblocking( 0 )
+    try:
+      osSocket = socket.socket( sockType, socket.SOCK_STREAM )
+    except socket.error as e:
+      gLogger.warn( "Exception while creating a socket:", str( e ) ) 
+      return S_ERROR( "Exception while creating a socket:%s" % str( e ) )
```

Comment:
Just as a comment, here, str is not needed because that is what %s will already do